python/src/grid.cpp

```diff
@@ -54,19 +54,23 @@ void add_grid_bindings(py::module& m) {
     pgrid.def(py::init<profile_grid::size_type, profile_grid::spacing_type, const profile_grid::boundary_status_type&, const std::vector<fs::node>>());
     pgrid.def(py::init(
                 [](std::size_t size,
-                    double spacing,
+                    profile_grid::spacing_type spacing,
                     const std::array<fs::node_status, 2>& bs,
                     const std::vector<std::pair<std::size_t, fs::node_status>>& ns)
-                 {
-                     std::vector<fs::node> node_vec;
-                     for (auto&& node : ns)
-                     {
-                         node_vec.push_back({node.first, node.second});
-                     }
-                     return std::make_unique<profile_grid>(size, spacing, bs, node_vec);}));
-    
+                {
+                    std::vector<fs::node> node_vec;
+                    for (auto&& node : ns)
+                    {
+                        node_vec.push_back({node.first, node.second});
+                    }
+                    return std::make_unique<profile_grid>(size, spacing, bs, node_vec);
+                }));
+
+    pgrid.def_static("from_length", &profile_grid::from_length);
```

Comment:
Perhaps we could [bind this factory as a constructor](https://pybind11.readthedocs.io/en/stable/advanced/classes.html?highlight=factory#custom-constructors), which is a bit more Python idiomatic.

Reply:
not sure it's possible, for the same reason we needed to add a static method with a factory pattern : the constructor and this method have the same signature.
I will try :)

Reply:
Ah, right. No big deal if that's not possible.

Reply:
There is no compilation error but only the regular cpp constructor is used.